trace/trace_test.go

```diff
@@ -446,6 +446,30 @@ func TestMessageEvents(t *testing.T) {
 	}
 }
 
+func TestSetSpanName(t *testing.T) {
+	span := startSpan(StartOptions{})
+	span.SetName("NewName")
+	got, err := endSpan(span)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := &SpanData{
+		SpanContext: SpanContext{
+			TraceID:      tid,
+			SpanID:       SpanID{},
+			TraceOptions: 0x1,
+		},
+		ParentSpanID:    sid,
+		Name:            "NewName",
+		Status:          Status{},
+		HasRemoteParent: true,
+	}
+	if !reflect.DeepEqual(got, want) {
```

Comment:
Just check the name here to make it clear what you are testing.

Reply:
Also would be good to test what happens if you call SetName when not collecting events and also after the Span has ended.

Reply:
sure